### PR TITLE
headless: Add some new and passing tests

### PR DESCRIPTION
**Pull request by unknownbrackets**
Also small fix to an existing one to make it pass as well.

-[Unknown]

### CODE REVIEW

File: GPU/Common/VertexDecoderArm.cpp [hunk 1]
```diff
@@ -872,22 +872,14 @@ void VertexDecoderJitCache::Jit_NormalFloat() {
...
 	_dbg_assert_msg_(fpScratchReg + 1 == fpScratchReg2, "VertexDecoder fpScratchRegs must be in order.");
 	_dbg_assert_msg_(fpScratchReg2 + 1 == fpScratchReg3, "VertexDecoder fpScratchRegs must be in order.");
 
-	// TODO: SIMD
-	LDRSB(tempReg1, srcReg, dec_->posoff);
-	LDRSB(tempReg2, srcReg, dec_->posoff + 1);
-	LDRB(tempReg3, srcReg, dec_->posoff + 2);
-	static const ARMReg tr[3] = { tempReg1, tempReg2, tempReg3 };
-	static const ARMReg fr[3] = { fpScratchReg, fpScratchReg2, fpScratchReg3 };
+	// Intentionally just sets all to zero.
```
Comment (hrydgard) [inline GPU/Common/VertexDecoderArm.cpp:879]: I think a better comment here is what you have below:

```
// 8-bit positions in throughmode always decode to 0, depth included.
```

File: GPU/Common/VertexDecoderArm64.cpp [hunk 2]
```diff
@@ -668,15 +668,11 @@ void VertexDecoderJitCache::Jit_PosFloat() {
 }
 
 void VertexDecoderJitCache::Jit_PosS8Through() {
-	LDRSB(INDEX_UNSIGNED, tempReg1, srcReg, dec_->posoff);
-	LDRSB(INDEX_UNSIGNED, tempReg2, srcReg, dec_->posoff + 1);
-	LDRB(INDEX_UNSIGNED, tempReg3, srcReg, dec_->posoff + 2);
-	fp.SCVTF(fpScratchReg, tempReg1);
-	fp.SCVTF(fpScratchReg2, tempReg2);
-	fp.SCVTF(fpScratchReg3, tempReg3);
+	// Intentionally just sets all to zero.
```
Comment (hrydgard) [inline GPU/Common/VertexDecoderArm64.cpp:671]: Same

File: GPU/Common/VertexDecoderArm.cpp [hunk 3]
```diff
@@ -872,22 +872,14 @@ void VertexDecoderJitCache::Jit_NormalFloat() {
 	STMIA(scratchReg, false, 3, tempReg1, tempReg2, tempReg3);
 }
 
-// Through expands into floats, always. Might want to look at changing this.
 void VertexDecoderJitCache::Jit_PosS8Through() {
```
Comment (hrydgard) [inline GPU/Common/VertexDecoderArm.cpp:875]: Should this be renamed PosInvalid too?

**Review by hrydgard — CHANGES_REQUESTED** [review]

File: GPU/Common/VertexDecoderArm.cpp (hunk 3 above)
Comment (unknownbrackets) [reply GPU/Common/VertexDecoderArm.cpp:875]: Well, "invalid" I used for when no position is specified at all, and the verts are all culled.

However, when a position is specified, as s8 in through, it is not culled.  It decodes to zero.  So it's kinda invalid, but different from "invalid position in throughmode (no bits set)" which has different behavior.

-[Unknown]

Comment (unknownbrackets) [reply GPU/Common/VertexDecoderArm.cpp:875]: As to Arm/Arm64/etc. - I figured might as well keep these jtited, although they're hopefully rarely being hit.  I didn't think it was worth bothering to newly jit invalid, though.

-[Unknown]

Comment (hrydgard) [reply GPU/Common/VertexDecoderArm.cpp:875]: Ok, cool. Thanks for the clarifications!

**Review by hrydgard — APPROVED** [review]

**Comment by hrydgard** [comment]
Feel free to merge if it's ready :)

**Pull request merged**